PLANET-7610: Rename the Submenu block as Table Of Contents

mardelnet opened the pull request:
Ref: https://jira.greenpeace.org/browse/PLANET-7610

<hr>

**DESCRIPTION:**
As the former "Submenu" block was renamed as "Table of Contents", this PR replaces every mention of the "submenu" word in our codebase, including strings, variable names, file names, constant values, etc. However, to avoid data loss, the actual name of the block (`submenu`) is kept until the migration of this block is completed.

Additionally, this PR introduces a file (`assets/src/blocks/Constants.js`) to store and handle the blocks' constant values, to avoid duplication and keep the code cleaner. This file should be updated in further PRs.

<hr>

**RELATED PR:**
https://github.com/greenpeace/planet4-master-theme/pull/2376

mardelnet committed:
PLANET-7610: Rename the Submenu block

- The Submenu block was renamed as Table of Contents

planet-4 commented:
### Test instance is ready :rocket:

:new_moon: [umbriel](https://www-dev.greenpeace.org/test-umbriel) | [admin](https://www-dev.greenpeace.org/test-umbriel/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-umbriel/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-umbriel) | [composer-local.json](https://github.com/greenpeace/planet4-test-umbriel/blob/main/composer-local.json)

:watch: 2024.09.10 14:33:07

mardelnet committed:
PLANET-7610: Rename the Submenu block

- Revert changes in translation files

mardelnet committed:
PLANET-7610: Rename the Submenu block

- Revert changes in the block name to avoid problems related to block migration

mardelnet committed:
PLANET-7610: Rename the Submenu block

- Fix lint issues

mardelnet commented:
Closing this PR in favor of https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/1242
